src/cunumeric/index/zip.cu

```diff
@@ -83,10 +85,63 @@ __global__ static void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
   out[p] = new_point;
 }
 
+template <typename Output, int DIM>
+__global__ static void __launch_bounds__(THREADS_PER_BLOCK, MIN_CTAS_PER_SM)
+  check_kernel(Output out,
+               const Buffer<AccessorRO<int64_t, DIM>, 1> index_arrays,
+               const int64_t volume,
+               const int64_t iters,
+               const Rect<DIM> rect,
+               const Pitches<DIM - 1> pitches,
+               const int64_t narrays,
+               const int64_t start_index,
+               const DomainPoint shape)
+{
+  bool value = false;
+  for (size_t i = 0; i < iters; i++) {
+    const auto idx = (i * gridDim.x + blockIdx.x) * blockDim.x + threadIdx.x;
+    if (idx >= volume) break;
+    auto p = pitches.unflatten(idx, rect.lo);
+    for (size_t n = 0; n < narrays; n++) {
+      const int64_t extent = shape[start_index + n];
+      coord_t index = index_arrays[n][p] < 0 ? index_arrays[n][p] + extent : index_arrays[n][p];
+      bool val      = (index < 0 || index > extent);
+      SumReduction<bool>::fold<true>(value, val);
+    }  // for n
+  }
+  reduce_output(out, value);
+}
+
 template <int DIM, int N>
 struct ZipImplBody<VariantKind::GPU, DIM, N> {
   using VAL = int64_t;
 
+  void check_out_of_bounds(const Buffer<AccessorRO<int64_t, DIM>, 1>& index_arrays,
+                           const int64_t volume,
+                           const Rect<DIM>& rect,
+                           const Pitches<DIM - 1>& pitches,
+                           const int64_t narrays,
+                           const int64_t start_index,
+                           const DomainPoint& shape,
+                           cudaStream_t stream) const
+  {
+    const size_t blocks = (volume + THREADS_PER_BLOCK - 1) / THREADS_PER_BLOCK;
+    size_t shmem_size   = THREADS_PER_BLOCK / 32 * sizeof(int64_t);
```

Comment:
The kernel is reducing to a value of type `bool`, so it should be `sizeof(bool)` here.